force-app/main/default/classes/primaryContactExtension.cls

```diff
@@ -0,0 +1,69 @@
+public with sharing class primaryContactExtension {
+
+    private final Account acct;
+    public String searchVar{get; set;}
+    public boolean displayPopup{get; set;}
+    public String primContactId{get; set;}
+
+    // The extension constructor initializes the private member
+    // variable acct by using the getRecord method from the standard
+    // controller.
+    public primaryContactExtension(ApexPages.StandardController stdController) {
+        this.acct = (Account)stdController.getRecord(); 
+    }
+
+    public List<Contact> getRelatedContacts() {
+        if (searchVar == null || searchVar == '') {
+            return [SELECT Id, Name, Title, Phone, Is_Primary_Contact__c FROM Contact WHERE AccountId=:acct.Id Limit 1000];
+        }
+        else {
+            String escapedString = string.escapeSingleQuotes(searchVar);
+            escapedString = '%' + escapedString + '%';
+            return [SELECT Id, Name, Title, Phone, Is_Primary_Contact__c FROM Contact WHERE AccountId=:acct.Id AND Name like :escapedString Limit 1000];
+        }
+    }
```

Comment:
All dml operations should be wrapper with try catch block and instead of limit, please use pagination

Reply:
try catch applies all occurances

Reply:
try catch blocks added for every dml/soql operations and added pagination